src/Castle.Windsor/Core/Internal/ReflectionUtil.cs

```diff
@@ -150,14 +150,20 @@ public static Assembly GetAssemblyNamed(string assemblyName)
 
 		public static Assembly[] GetLoadedAssemblies()
 		{
-#if SILVERLIGHT
-			var list =
-				System.Windows.Deployment.Current.Parts.Select(
-					ap => System.Windows.Application.GetResourceStream(new Uri(ap.Source, UriKind.Relative))).Select(
-						stream => new System.Windows.AssemblyPart().Load(stream.Stream)).ToArray();
-			return list;
-#else
+#if FEATURE_APPDOMAIN
 			return AppDomain.CurrentDomain.GetAssemblies();
+#else
+			string basePath = AppContext.BaseDirectory;
+
+			var files = new DirectoryInfo(basePath).GetFiles("*.dll");
+			Assembly[] results = new Assembly[files.Length];
+			
+			for(int i = 0; i < files.Length; i++)
+			{
+				results[i] = System.Runtime.Loader.AssemblyLoadContext.Default.LoadFromAssemblyPath(files[i].FullName);
```

Comment:
It looks like there aren't any options to get these assemblies without loading them. The number of loaded assemblies can be reduced by filtering the names with `ShouldSkipAssembly()` before loading. Checking `AssemblyName.ContentType` can help filter out the native assemblies.
I have pushed a commit that uses `DependencyContext.RuntimeLibraries` to get the assemblies. It works with .NET Standard 1.6. To support .NET Standard 1.5, the way to get these assemblies would be still to look in the base directory for dll's, or if the application assembly is passed in, to get them with  `Assembly.GetReferencedAssemblies()`